### ISSUE TITLE
Cannot load stats 

### ISSUE THREAD

**Issue by scottiesan** [Steem-FOSSbot/steem-fossbot-voter#129]
Seems to be timing out.  

2018-02-11T03:06:47.716689+00:00 heroku[router]: at=error code=H12 desc="Request timeout" method=GET path="/stats" host=steem-power.herokuapp.com request_id=2aded27a-0cb2-4b15-a9ab-ffa92d7ad8a2 fwd="223.141.99.45" dyno=web.1 connect=0ms service=30001ms status=503 bytes=0 protocol=https

Any work around for this?   Thanks!

**Comment by thrize** [comment]
Not really, that's not a whole lot of information. Can you give me a more detailed log?

**Comment by jeaimetu** [comment]
I have same issue. On first running, there was no error. However after a couple of running, this error occurs when I want to enter "stat" menu. and error message was same.

**Comment by jeaimetu** [comment]
To workaround this, I deleted MongoDB except configuration and algorithm. Then I can enter bot stats. I think I changed configuration counter of max read is the reason of this. By the way, who decide the "request time out parameter"? Am I changed this to modify lib.js source code?

After deletion of some DB and restart FOSSBOT, then Bot said as follows on entering stat.

No stats available
It looks like this is a fresh install of Voter. Please generate some stats by using it and then come back here to see the results in detail.

Thanks.

**Comment by thrize** [comment]
I can also confirm I can replicate. This is another knock on due to the bot having trouble scaling to the new traffic of Steemit posts.

The stats will have to be segmented in some way which will require a bit of UI re-design also.

**Comment by thrize** [comment]
The solution I've implemented here is to cap the amount of data recorded to a reasonable amount instead of recording all the information on all posts processed. It is just too much to be displayed and is not useful anyway. There is not very sophisticated data processing in the front end for the web app (it really needs to be improved, but that's for another day) so I think this solution is good.

I've added two new config vars, with the following defaults:

* `POST_METADATA_MAX_RECORD_PER_RUN`: `100`
* `POST_METADATA_MAX_RUNS_TO_KEEP`: `5`

The first is the number of posts to record per bot run. If you have `MAX_POST_TO_READ` set to `1000` then it will only record the first 10% of them.

The second is the number of bot runs to keep. This is already restricted by date using `DAYS_KEEP_LOGS` but it further restricts it to the last number that you specify. It is set to a low value because it is assumed that you only care about review the last few when seeing if your algorithm is performing as you expect.

There are a couple other changes to make to the docs which I'll do now before passing this on for review.
